framework/src/engine/metrics/metrics.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+import { Counter, Gauge, Histogram, Registry } from 'prom-client';
+
+const noopCounter = {
+	inc: (_num?: number) => {},
+	reset: () => {},
+};
+
+const noopGauge = {
+	inc: (_num?: number) => {},
+	dec: (_num?: number) => {},
+	set: (_num: number) => {},
+};
+
+const noopHistogram = {
+	observe: (_num: number) => {},
+	startTimer: () => () => {},
+	reset: () => {},
+};
+
+export class Metrics {
+	private _enable: boolean;
```

Comment:
A small semantic comment:

I personally avoid to name any variables as verbs, as verbs should rather be used for function names.

I think that a variable that describes a state should always be an adjective: `enabled`, while a function that mutates the state should be verb: `enable()` / `disable()`.